packages/common/src/error.ts

```diff
@@ -11,93 +11,147 @@
-        return translateOrFallback(this.tKey, this.tParams, this.message, i18n)
-    }
-}
+    constructor(error: Error | TranslationKey, options?: ErrorOptions) {
+        const logger = options?.logger || getLoggerDefault()
+        const logLevel = options?.logLevel || 'error'
 
-export class ProsopoEnvError extends ProsopoBaseError {
-    cause: Error | undefined
-
-    constructor(error: ProsopoBaseError)
-    constructor(error: Error, context?: TranslationKey, options?: TOptions, ...params: any[])
-    constructor(error: TranslationKey, context?: string, options?: TOptions, ...params: any[])
-
-    constructor(
-        error: Error | TranslationKey,
-        context?: TranslationKey | string,
-        options?: TOptions,
-        ...params: any[]
-    ) {
-        const isError = error instanceof Error
-        super(isError ? error.message : translateOrFallback(error, options))
-        this.name =
-            (context && `${ProsopoEnvError.name}@${translateOrFallback(context, options)}`) || ProsopoEnvError.name
-        if (isError) {
-            this.cause = error
-            if (error instanceof ProsopoBaseError) {
-                this.tKey = error['tKey']
-                this.tParams = error['tParams']
-            } else if (translationKeys.includes(context as TranslationKey)) {
-                this.tKey = context as TranslationKey
-                this.tParams = options || {}
+        // Check if error is an Error
+        if (error instanceof Error) {
+            super(error.message)
+            this.translationKey = options?.translationKey
+            this.context = options?.context
+            if (options?.translationKey) {
+                this.context = { ...this.context, translationMessage: i18next.t(options.translationKey) }
             }
         } else {
-            this.tKey = error as TranslationKey
-            this.tParams = options || {}
+            super(i18next.t(error))
+            this.translationKey = error
+            this.context = options?.context
         }
 
-        console.error('\n********************* ERROR *********************\n')
-        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-        // @ts-ignore
-        if (typeof this.cause != ZodError && this.cause?.message && this.cause.message.length > MAX_ERROR_LENGTH) {
-            this.cause.message = `${at(this.cause.message, -MAX_ERROR_LENGTH)}...`
+        this.logError(logger, logLevel)
+    }
+
+    private logError(logger: Logger, logLevel: LogLevel) {
+        const errorFormatter = '\n********************* ERROR *********************\n'
+        const errorName = `Error Type: ${this.name}\n`
+        const errorParams = JSON.stringify({ error: this.message, context: this.context })
+        const errorMessage = `${errorFormatter}${errorName}${errorParams}`
+
+        switch (logLevel) {
+            case 'trace':
+                logger.trace(errorMessage, this.stack)
```

Comment:
We'd have to impl this in the logger.ts file, which would effectively just be moving the switch statement there. Didn't want to do this as I was concerned it would mess with previous usage of logger.log

Reply:
https://github.com/prosopo/captcha/pull/983

Reply:
clean fix nice

Reply:
addressed in latest push